### PR TITLE
Fixed bad error message with a poor marker.

### PR DESCRIPTION
**Pull request by pelson**
Found during @WeatherGod's scipy'14 tutorial.

### CODE REVIEW

**Comment by pelson** [comment]
I've updated - let's hope it passes now 6 months on :wink:

**Comment by tacaswell** [comment]
@pelson This needs a rebase

**Comment by pelson** [comment]
Let's see how that does.

**Comment by WeatherGod** [comment]
weird error with py2.6 (can't find the test_lines module...). For py3k builds, we got a string comparison issue ('7' != b'7').

File: lib/matplotlib/markers.py [hunk 1]
```diff
@@ -243,9 +243,9 @@ def set_marker(self, marker):
             try:
                 Path(marker)
                 self._marker_function = self._set_vertices
-            except ValueError:
-                raise ValueError('Unrecognized marker style'
-                                 ' {0}'.format(marker))
+            except (ValueError, AssertionError):
+                raise ValueError('Unrecognized marker style "{0}"'
+                                 ''.format(marker.encode('ascii', 'ignore')))
```
Comment (tacaswell) [inline lib/matplotlib/markers.py]: I think drop the 'encode' here

File: lib/matplotlib/tests/test_lines.py [hunk 2]
```diff
@@ -133,5 +133,12 @@ def test_nan_is_sorted():
     assert_true(not line._is_sorted(np.array([1,np.nan,3])))
 
 
+def test_invalid_marker():
+    # Specifying an invalid marker should raise a good exception.
+    line = mpl.lines.Line2D([], [])
+    assert_raises_regexp(ValueError, "Unrecognized marker style \"7\"",
+                         line.set_marker, u'7')
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_lines.py]: and drop the 'u' here?

**Comment by tacaswell** [comment]
Those usually mean sneaky import errors with nose.

File: lib/matplotlib/tests/test_lines.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 import six
 
 import nose
-from nose.tools import assert_true, assert_raises
+from nose.tools import assert_true, assert_raises, assert_raises_regexp
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_lines.py]: I bet that `assert_raises_regexp` is not available in 2.6.

File: lib/matplotlib/tests/test_lines.py (hunk 2 above)
Comment (pelson) [inline lib/matplotlib/tests/test_lines.py]: Inclined to agree. I was hoping to actually test the unicode definition, but hey-ho, it is harder to test.

File: lib/matplotlib/tests/test_lines.py (hunk 3 above)
Comment (pelson) [inline lib/matplotlib/tests/test_lines.py]: When are we dropping 2.6? We all know it is a dead release, is there a platform we are holding out for? RHEL5 defaults to python2.4, RHEL6 defaults to python2.6. If somebody wants to install the latest matplotlib, is it not reasonable to expect them to also install a version of python which is <7 years old....?

Comment (jenshnielsen) [inline lib/matplotlib/tests/test_lines.py]: Im :+1:  on dropping python 2.6. I think the main reason for keeping it around is indeed RHEL6

Comment (tacaswell) [inline lib/matplotlib/tests/test_lines.py]: I think dropping 2.6 would put us at the leading edge of the de-adoption curve, we sure we want to be there?

Also raises the question of when we can drop 2.7 as well :smiling_imp:

Comment (jenshnielsen) [inline lib/matplotlib/tests/test_lines.py]: Not sure that it's relevant but IPython dropped support some time ago with version 2.0

More to the point I guess we can maintain support for it as long as we can test it on Travis and the work arounds are limited to minor issues like this.

Comment (WeatherGod) [inline lib/matplotlib/tests/test_lines.py]: I can see keeping py2.6 for just a short while longer (maybe just for the
upcoming couple of releases?). But, I don't see 2.7 going away anytime
soon, unfortunately. We probably won't be one of those straggler projects
that holds on until 2020, but with the exception of keyword-only arguments,
I haven't seen much in py3k that is relevant to this project. There are
lots of goodies in py3k, I just haven't seen anything yet that made me want
to go "ooh, I really want that for feature X in matplotlib!".

**Comment by dstansby** [comment]
Since matplotlib no longer requires 2.6 could this be rebased and then we can run the tests and see what happens?

**Comment by anntzer** [comment]
As far as I can see this PR is no longer necessary (there's no way to trigger an AssertionError via a bad marker, and the error message is always reasonably formatted (well, we change `{0}` to `0!r` but that's rather minor).  Feel free to reopen if you disagree.

**Pull request closed without merging**